code/implementation/system-stack/src/main/java/io/cattle/platform/systemstack/listener/SystemStackUpdate.java

```diff
@@ -34,15 +36,18 @@
 
 import org.apache.commons.io.IOUtils;
 import org.apache.commons.lang3.StringUtils;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.client.fluent.Response;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class SystemStackUpdate implements AnnotatedEventListener {
 
-    private static final String[] STACKS = new String[] {"swarm", "kubernetes"};
+    private static final String[] STACKS = new String[] {"swarm", "kubernetes", "publicdns"};
     private static final String STACK_EXTERNAL_ID = "system://%s";
     private static final String STACK_RESOURCE = "/config-content/system-stacks/%s/%s";
     private static final Logger log = LoggerFactory.getLogger(SystemStackUpdate.class);
+    private static final String CATALOG_RESOURCE_URL = "http://localhost:8088/v1-catalog/templates/library:%s";
```

Comment:
We should probably make this a setting even though the web.xml hard codes it.  I'd call the setting something like `system.stack.catalog.url`.